test/test_jit.py

```diff
@@ -6801,6 +6801,24 @@ def foo2(a : Optional[int], b : Optional[int]):
         self.assertEqual(foo2(None, 4), 0)
         self.assertEqual(foo2(4, None), 0)
 
+        @torch.jit.script
+        def any_refinement(a, b):
+            # type: (Any, Any) -> int
+            if isinstance(a, int) and isinstance(b, int):
+                return a + b
+            return 0
+
+        self.assertEqual(any_refinement(3, 4), 7)
+        self.assertEqual(any_refinement(3, "hi"), 0)
+
+    def test_any_in_class_fails(self):
```

Comment:
Can we add tests that verify that `Any` is not allowed in Tuples, NamedTuples, or Interfaces as well?